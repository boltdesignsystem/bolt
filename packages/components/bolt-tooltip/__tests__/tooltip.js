import { render, stopServer } from '../../../testing/testing-helpers';
import schema from '../tooltip.schema';
const { placement } = schema.properties;

describe('<bolt-tooltip> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  // Basic Usage
  test('Basic usage', async () => {
    const results = await render('@bolt-components-tooltip/tooltip.twig', {
      trigger: 'CRM',
      content: 'Customer relationship management',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Advanced Usage
  test('Advanced usage: adding tooltip to button', async () => {
    const results = await render('@bolt-components-tooltip/tooltip.twig', {
      trigger: '<button type="button" class="e-bolt-button">Download</button>',
      content: 'File size: 25MB',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  placement.enum.forEach(async placementChoice => {
    test(`Placement of the tooltip bubble: ${placementChoice}`, async () => {
      const results = await render('@bolt-components-tooltip/tooltip.twig', {
        trigger: 'CRM',
        content: 'Customer relationship management',
        placement: placementChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  test(`UUID of the tooltip`, async () => {
    const results = await render('@bolt-components-tooltip/tooltip.twig', {
      trigger: 'CRM',
      content: 'Customer relationship management',
      uuid: 'custom-unique-id',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
