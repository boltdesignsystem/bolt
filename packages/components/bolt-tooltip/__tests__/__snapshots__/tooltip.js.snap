// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-tooltip> Component Advanced usage: adding tooltip to button 1`] = `
<bolt-tooltip uuid="12345">
  <button type="button"
          class="e-bolt-button"
  >
    Download
  </button>
  <span slot="content">
    File size: 25MB
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Basic usage 1`] = `
<bolt-tooltip uuid="12345">
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: auto 1`] = `
<bolt-tooltip placement="auto"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: bottom 1`] = `
<bolt-tooltip placement="bottom"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: bottom-end 1`] = `
<bolt-tooltip placement="bottom-end"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: bottom-start 1`] = `
<bolt-tooltip placement="bottom-start"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: left 1`] = `
<bolt-tooltip placement="left"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: left-end 1`] = `
<bolt-tooltip placement="left-end"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: left-start 1`] = `
<bolt-tooltip placement="left-start"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: right 1`] = `
<bolt-tooltip placement="right"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: right-end 1`] = `
<bolt-tooltip placement="right-end"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: right-start 1`] = `
<bolt-tooltip placement="right-start"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: top 1`] = `
<bolt-tooltip placement="top"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: top-end 1`] = `
<bolt-tooltip placement="top-end"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: top-start 1`] = `
<bolt-tooltip placement="top-start"
              uuid="12345"
>
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component UUID of the tooltip 1`] = `
<bolt-tooltip uuid="custom-unique-id">
  CRM
  <span slot="content">
    Customer relationship management
  </span>
</bolt-tooltip>
`;
